src/main/java/seedu/address/logic/commands/AddAssignmentToAllCommand.java

```diff
@@ -50,17 +56,42 @@ public AddAssignmentToAllCommand(Module module, Assignment assignment) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         // Get Person that match the Module
-        List<Person> filteredPersonList =
+        List<Person> personListInModule =
                 model.getAddressBook().getPersonList()
                         .stream()
                         .filter(person -> person.hasModule(module))
                         .collect(Collectors.toList());
 
-        if (filteredPersonList.size() == 0) {
+        if (personListInModule.size() == 0) {
             throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_MODULE);
         }
 
-        model.addAllAssignment(filteredPersonList, toAdd);
+        //Split the persons into those with or without the assignment
+        List<List<Person>> filteredPersonList = new ArrayList<>(personListInModule
+                .stream()
+                .collect(Collectors.partitioningBy(person -> model.hasAssignment(person, toAdd))).values());
+
+        List<Person> personListWithoutAssignment = filteredPersonList.get(0);
+        List<Person> personListWithAssignment = filteredPersonList.get(1);
+
+        Assignment existingAssignment = toAdd;
+        if (!personListWithAssignment.isEmpty()) {
```

Comment:
Perhaps wrap all this code up in a method and give it a suitable name? Method a bit long

Reply:
Will do!